src/modals/storage-connections/StorageConnectionsModal.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import PluginApiModal from '_/components/modals/PluginApiModal'
+import getPluginApi from '_/plugin-api'
+import { msg } from '_/intl-messages'
+import { Spinner } from 'patternfly-react'
+import { Button } from '@patternfly/react-core'
+import StorageConnectionsModalBody from './StorageConnectionsModalBody'
+import { createErrorMessage } from '_/utils/error-message'
+import './connections.css'
+
+const StorageConnectionsModal = ({
+  isLoading = false,
+  storageDomain,
+  connections,
+  onClose,
+  doConnectionCreate = () => { },
+  doConnectionEdit = () => { },
+  doConnectionDelete = () => { },
+  doConnectionAttach = () => { },
+  doConnectionDetach = () => { },
+  doRefreshConnections = () => { },
+}) => {
+  const [isOpen, setOpen] = useState(true)
+  const [error, setError] = useState(null)
+  const [isShowAll, setShowAll] = useState(false)
```

Comment:
In the current version it still feels in the wrong place. It should go into`StorageConnectionsModalBody`   next to
`const [isNewConnectionOpened, setNewConnectionOpened] = useState(false)`